scripts/top-search-queries/get-top-search-results.go

```diff
@@ -0,0 +1,242 @@
+	"flag"
+	"fmt"
+	"io"
+
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ONSdigital/dp-search-api/models"
+	"github.com/ONSdigital/dp-search-api/sdk"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// A searchResponse struct to map the response, from the Search API, to a particular query
+type searchResponse struct {
+	Items []searchItem `json:"items"`
+}
+
+// A SearchItem Struct to map each searchItem of the searchResponse to.
+type searchItem struct {
+	Type        string `json:"type"`
+	Edition     string `json:"edition"`
+	ReleaseDate string `json:"release_date"`
+	Summary     string `json:"summary"`
+	Title       string `json:"title"`
+	Uri         string `json:"uri"`
+}
+
+type runConfig struct {
+	APIURL        string `json:"apiurl"`
+	InputFileName string `json:"inputFileName"`
+	NumResults    int    `json:"numResults"`
+}
+
+// main reads in a list of queries from a text file. These are intended to be the queries that are most commonly used
+// in the live Search API service.
+//
+// It then creates an output CSV file and processes the queries, by calling the live Search API, and outputs
+// all the query results to that CSV file.
+//
+// Each query is called with NLP (Natural Language Processing) initially switched off, then called again with
+// NLP switched on. The top 10 results of each query call are added to the output CSV file.
+func main() {
+	ctx := context.Background()
+	log.Info(ctx, "starting script to get results of most common queries from the Search API")
+
+	config := runConfig{}
+	flag.StringVar(&config.APIURL, "api_url", "https://api.beta.ons.gov.uk/v1", "the base url for the search api")
+	flag.StringVar(&config.InputFileName, "input_file_name", "search-queries.txt", "name of input file including extension")
+	flag.IntVar(&config.NumResults, "num_results", 10, "number of results to fetch for each query")
+	flag.Parse()
+	logData := log.Data{"config: ": config}
+	log.Info(ctx, "parsed config", logData)
+	run(ctx, config)
+	log.Info(ctx, "end of script")
+}
+
+func run(ctx context.Context, config runConfig) {
+	listOfQueries, err := readQueriesFromFile(ctx, config)
+	check(ctx, "failed reading queries file", err)
+
+	csvFile := createOutputCSVFile(ctx)
+	defer func() {
+		if err = csvFile.Close(); err != nil {
+			log.Fatal(ctx, "failed closing csv file", err)
+		}
+	}()
+
+	log.Info(ctx, "make each request to the Search API with NLP off initially, then NLP on")
+	for _, query := range listOfQueries {
+		searchResp := getQueryResults(ctx, query, "false", config)
+		AddResultsToCSV(ctx, query, csvFile, "false", searchResp)
+		searchResp = getQueryResults(ctx, query, "true", config)
+		AddResultsToCSV(ctx, query, csvFile, "true", searchResp)
+	}
+}
+
+// createOutputCSVFile creates the CSV file ready for the query results to be added to. It firstly adds a header row.
+func createOutputCSVFile(ctx context.Context) *os.File {
+	logData := log.Data{"CSV name: ": "top-search-query-results.csv"}
```

Comment:
There are a number of log lines in the script that don't follow [our logging standards](https://github.com/ONSdigital/dp-standards/blob/main/LOGGING_STANDARDS.md#logging-specification).
- events should convey what has happened in the code at that point- ie. what 'event' just happened
- log data has a key and a value, the key should use underscores and have no spaces or punctuation
```suggestion
	logData := log.Data{"output_filename": "top-search-query-results.csv"}
```